step-capstone/src/components/AddEvent.js

```diff
@@ -34,6 +34,18 @@ export default function AddEvent(props) {
         setDescription(e.target.value);
     }
 
+    useEffect(() => {
```

Comment:
Maybe put these in the same useEffect since they both happen at the same time?

Reply:
Also I thought we took this out? From my understanding, useEffect is used when component states are changed. If props.data changes the entire function would be re-rendered anyways since it would be as state change in the parent component (form) which automatically triggers a re-render of this component and any other Add component.

Reply:
Hi Emmie, so this is the short answer for why: https://stackoverflow.com/questions/38892672/react-why-child-component-doesnt-update-when-prop-changes#:~:text=Because%20children%20do%20not%20rerender,if%20its%20STATE%20changes%20%3A)&text=It%20will%20pass%20data%20from,is%20no%20rerender%20logic%20there.&text=According%20to%20React%20philosophy%20component,parent%20and%20should%20be%20immutable.
Here is one experiment from our app:
![image](https://user-images.githubusercontent.com/42331170/87161307-5a0c7100-c292-11ea-94b7-aef4babec507.png)

You could see that in the left, ItemForm rerender with 3AM - 6AM (start - end), and the new props in AddEvent also shows 3 AM - 6 AM. But the form, in reality, doesn't update the state. That's why I added these useEffect to listen to the change and update the state.